### PR TITLE
chore: Remove flipper pods from RN example app

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

Remove Flipper pods and ensure Hermes engine is used when building iOS example app for faster pods install and app build both locally and in CI.
Small gemfile changes as per recommendation from RN migration helper tool.

While working on re-enabling tests in CI ran into issues where iOS app wasn't building or was taking 30-40min to build in CI, breaking some of the changes out. Ref draft  [PR](https://github.com/aws-amplify/amplify-ui/pull/4785/files).

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/ios/Podfile [hunk 1]
```diff
@@ -13,7 +13,8 @@ prepare_react_native_project!
 #   dependencies: {
 #     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
 # ```
-flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
+#flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
```
Comment (calebpollman) [inline examples/react-native/ios/Podfile:16]: Should we delete this line entirely?

Comment (ioanabrooks) [reply examples/react-native/ios/Podfile:16]: Yeah sure, I'm on board, don't think there's a world where we build with flipper again :)

Comment (ioanabrooks) [inline examples/react-native/ios/Podfile:16]:
```suggestion
```

**Review by calebpollman — APPROVED** [review]
🚢

**Review by dbanksdesign — APPROVED** [review]

**Pull request merged**